bibliopixel/control/api.py

```diff
@@ -0,0 +1,56 @@
+from http.server import HTTPServer, BaseHTTPRequestHandler
+import urllib
+import getpass, platform, sys, threading
+from .. util import log
+from . control import ExtractedControl
+
+class Api(ExtractedControl):
+    EXTRACTOR = {
+        'keys_by_type': {
+            'direct_command': ['action'],
+            'value_command': ['action', 'value']
+        }
+    }
+
+    def make_value_commands(self, value_command_list):
+        return [{'action': com, 'value': val, 'type': 'value_command'} for (com, val) in value_command_list]
+
+    def make_direct_commands(self, direct_command_list):
+        if direct_command_list == None:
+            return []
+        else:
+            return [{'action': com, 'type': 'direct_command'} for (com) in direct_command_list]
+
+    def make_messages(self, params):
+        parsed = (urllib.parse.parse_qs(params[2:]))
```

Comment:
Q: lose the outer set of parentheses.

This doesn't use `self` and should be a free function.

You never use params[0] and params[1] so why did they get passed in in the first place?  :-D